datalad/distribution/dataset.py

```diff
@@ -384,16 +391,20 @@ def get_superdataset(self, topmost=False):
             # None was found
             return None
 
-        if realpath(self.path) != self.path:
-            # we had symlinks in the path but sds_path would have not
-            # so let's get "symlinked" version of the superdataset path
-            sds_relpath = relpath(sds_path, realpath(self.path))
-            sds_path = normpath(opj(self.path, sds_relpath))
+        # I think we might have accounted for it with use of normpath there??
```

Comment:
yes, it is a misleading wording and I think it is indeed irrelevant.  I do have some gut memory that somehow use of normpath here "normalized' behavior of `get_toppath` in some case, but now I am no longer have clear recollection. I will adjust comment to not mention normpath and will remove all the code leaving just a new comment hinting that post-processing should not be necessary.
Use of `normpath` imho there is warranted so we normalize the path after adding '..' and possibly avoiding some logic escaping into `..` of a symlinked directory into its original parent.